core/src/main/scala/org/apache/spark/ui/storage/StorageTab.scala

```diff
@@ -59,7 +59,15 @@ class StorageListener(storageStatusListener: StorageStatusListener) extends Bloc
 
   override def onStageSubmitted(stageSubmitted: SparkListenerStageSubmitted): Unit = synchronized {
     val rddInfos = stageSubmitted.stageInfo.rddInfos
-    rddInfos.foreach { info => _rddInfoMap.getOrElseUpdate(info.id, info) }
+    rddInfos.foreach { info =>
```

Comment:
@zsxwing That was my first attempt at this bug, but I've run into the problem described by @andrewor14 - there's a bunch of fields that get mutated in `onBlockUpdated` and, by just substituting `RDDInfo` object, we'll lose those changes. (The `BlockUpdated` event isn't triggered after name change, so we need to make sure we preserve these changed values.)